raven/src/main/java/org/opencadc/raven/RavenInitAction.java

```diff
@@ -173,6 +190,36 @@ private void initKeys() {
         log.info("initKeys: OK");
     }
 
+    private void initAvailabilityCheck() {
+        ArtifactDAO artifactDAO = new ArtifactDAO();
```

Comment:
Why create an ArtifactDAO here? The StorageSiteDAO copy ctor just copies the config map so you could instantiate it directly.

In principle, raven could run long enough that a new StorageSite appeared in the database. So the `list()` should get refreshed occasionally. So the list query should happen down in AvailabilityCheck.run(), right before the loop over known sites... doesn't need to happen every time, maybe same cadence of the full avail checks (5 min)

Reply:
Directly instantiated the StorageSiteDAO constructor, and moved the StorageSite query into the AvailabilityCheck run() method, where the query runs every AVAILABILITY_FULL_CHECK_TIMEOUT secs (currently 300).